src/parsing/parsed.rs

```diff
@@ -648,31 +655,44 @@ impl TryFrom<Parsed> for Date {
     }
 }
 
+fn naive_time_from_parsed(parsed: Parsed) -> Result<Time, error::TryFromParsed> {
```

Comment:
While I understand what you're going for here (and other similar locations), it would probably be best to reject the leap second outright for types other than `OffsetDateTime`, as they don't contain the necessary information in the final value to know if it was at a leap second or not.

Reply:
The information is available in the RFC 3339 source data to make that call. But if it isn't, e.g. when the `Parsed` is coming from another well-known format, then it's an "insufficient information" error. I agree this could be cryptic. Will redo as proposed.